e2e/auth.setup.ts

```diff
@@ -1,114 +1,66 @@
 // Required environment variables:
-// - ADMIN_USERNAME: username for admin
-// - ADMIN_PASSWORD: password for admin
+// - ADMIN_PAT: personal access token for admin
 //
 // Environment variables:
 // - in CI: set in repo GitHub secrets, read via env in workflow
 // - locally: set in .env file, read by dotenv
 
 import { Page, expect, test as setup } from '@playwright/test'
-import { v4 as uuidv4 } from 'uuid'
-import { ADMIN_STORAGE_STATE, USER_STORAGE_STATE } from '../playwright.config'
 import { getEndpoint } from './helpers/http'
 import { setLocalStorage } from './helpers/localStorage'
 import {
-  USER_NAME_LOCALSTORAGE_KEY,
+  cleanupTestUser,
   createTestUser,
-  deleteTestUser,
-  getAccessTokenFromCookie,
-  getAdminUserCredentials,
+  getAdminPAT,
   loginTestUser,
 } from './helpers/testUser'
-import { TestUser } from './helpers/types'
-
-let testUserId: string | undefined
-let adminAccessToken: string | undefined
-let adminPage: Page
-let userPage: Page
-
-setup.describe('Setup', () => {
-  setup.beforeAll(async ({ browser }) => {
-    adminPage = await browser.newPage()
-    userPage = await browser.newPage()
-  })
-
-  setup('authenticate users', async () => {
-    // This may be the first test to run and trigger SWC compilation
-    // ...which may take 15+ seconds. Mark this test as slow,
-    // ...so the test will be given triple timeout.
-    // https://playwright.dev/docs/api/class-test#test-slow-1
-    setup.slow()
-
-    await setup.step('authenticate admin', async () => {
-      const { adminUserName, adminUserPassword } = getAdminUserCredentials()
-
-      await loginTestUser(adminPage, adminUserName!, adminUserPassword!)
-      await adminPage.context().storageState({ path: ADMIN_STORAGE_STATE })
-
-      adminAccessToken = await getAccessTokenFromCookie(adminPage)
+import { userConfigs } from './helpers/userConfig'
+
+for (const {
+  testName,
+  storageStatePath,
+  localStorageKey,
+  user,
+} of userConfigs) {
+  let userId: string | undefined
+  let userPage: Page
+
+  setup.describe(`Setup: ${testName}`, () => {
```

Comment:
Loop through a config file to create and authenticate test users, which will hopefully make it easier to create new users in the future

Reply:
This is great! Easily extensible too to add controls like ACT status, certified, etc